File: examples-sdk/java/003-basic-hello-server/Main.java
// 003-basic-hello-server — Java SDK equivalent
// Compile: vil compile --from java --input 003-basic-hello-server/Main.java --release

import dev.vil.*;

public class Main {
    public static void main(String[] args) {
        VilServer server = new VilServer("vil-basic-hello-server", 8080);
        ServiceProcess gw = new ServiceProcess("gw");
        gw.endpoint("POST", "/transform", "transform");
        gw.endpoint("POST", "/echo", "echo");
        gw.endpoint("GET", "/health", "health");
        server.service(gw);
        server.compile();
    }
}

File: examples-sdk/swift/003-basic-hello-server/main.swift
// 003-basic-hello-server — Swift SDK equivalent
// Compile: vil compile --from swift --input 003-basic-hello-server/main.swift --release

let server = VilServer(name: "vil-basic-hello-server", port: 8080)
let gw = ServiceProcess(name: "gw")
gw.endpoint(method: "POST", path: "/transform", handler: "transform")
gw.endpoint(method: "POST", path: "/echo", handler: "echo")
gw.endpoint(method: "GET", path: "/health", handler: "health")
server.service(gw)
server.compile()
